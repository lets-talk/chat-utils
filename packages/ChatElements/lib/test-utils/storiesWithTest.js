import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests(jestTestResults, {
  filesExt: '__tests__/**.js',
});
