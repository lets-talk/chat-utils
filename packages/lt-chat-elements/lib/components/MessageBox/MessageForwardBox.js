import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

// Icons to be used on this component
import FaForward from 'react-icons/lib/fa/mail-forward';

const StyledMessageForwardBox = styled.div`
  display: flex;
  width: 30px;
  height: 30px;
  border-radius: 20px;
  background: #fff;
  position: absolute;
  flex-direction: row;
  align-self: center;
  align-items: center;
  justify-content: center;
  box-shadow: 0 0 5px 0 rgba(164, 164, 164, 1);
  cursor: pointer;
  transition: all 0.3s ease;
  top: 0;
  bottom: 0;
  margin: auto;

  ${(props) =>
    props.position === 'left' &&
      css`
        right: -50px
      `};

  ${(props) =>
    props.position === 'right' &&
      css`
        left: -50px
      `};
`;

const MessageForwardBox = (props) => (
  <StyledMessageForwardBox
    position={props.position}
    onClick={props.onForwardClick}
    onKeyPress={props.onForwardClick}
  >
    <FaForward color="#000" />
  </StyledMessageForwardBox>
);

MessageForwardBox.propTypes = {
  /**
   * Position: What side the forward is displayed
   */
  position: PropTypes.string,
  /**
   * onForwardClick: Handler function to be called when user click on forwared message
   */
  onForwardClick: PropTypes.func,
};

export default MessageForwardBox;
