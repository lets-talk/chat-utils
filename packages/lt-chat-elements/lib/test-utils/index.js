// test-utils/index.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const defaultTheme = {
  palette: {
    common: {
      black: 'rgb(0, 0, 0)',
      white: 'rgb(255, 255, 255)',
      online: 'rgb(67, 160, 71)',
      offline: 'rgb(229, 57, 53)',
      sleeping: 'rgb(253, 216, 53)',
    },
    colors: {
      primary: {
        base: 'rgb(4, 38, 73)',
        light: 'rgb(86, 165, 246)',
        dark: 'rgb(0, 0, 0)',
      },
      accent: {
        base: 'rgb(255, 87, 50)',
        light: 'rgb(255, 171, 152)',
        dark: 'rgb(203, 37, 0)',
      },
      foreground: {
        base: 'rgb(94, 124, 139)',
        light: 'rgb(151, 173, 184)',
        dark: 'rgb(53, 70, 78)',
      },
      background: {
        base: 'rgb(255, 255, 255)',
        light: 'rgb(255, 255, 255)',
        dark: 'rgb(204, 204, 204)',
      },
    },
    contrastColors: {
      dark: {
        primary: 'rgb(255, 255, 255)',
        secondary: 'rgba(255, 255, 255, 0.7)',
        disabled: 'rgba(255, 255, 255, 0.5)',
        hint: 'rgba(255, 255, 255, 0.12)',
      },
      light: {
        primary: 'rgba(0, 0, 0, 0.87)',
        secondary: 'rgba(0, 0, 0, 0.54)',
        disabled: 'rgba(0, 0, 0, 0.38)',
        hint: 'rgba(0, 0, 0, 0.12)',
      },
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    divider: 'rgba(255, 255, 255, 0.1)',
    action: {
      active: 'rgba(255, 255, 255, 0.9)',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.14)',
      disabled: 'rgba(255, 255, 255, 0.26)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)',
    },
  },
  shadows: {
    '2p': '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)',
    '3p': '0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 1px 8px 0 rgba(0, 0, 0, 0.12)',
    '4p': '0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.2)',
    '6p': '0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.2)',
    '8p': '0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 5px 5px -3px rgba(0, 0, 0, 0.2)',
    '16p': '0 16px 24px 2px rgba(0, 0, 0, 0.14), 0 6px 30px 5px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
  },
  typography: {
    fontFamily: "'Nimbus sans-serif', 'Sans', 'Helvetica', 'Arial'",
    fontSize: 'calc(1 * 1px)',
    weights: {
      fontWeightUltraThin: 100,
      fontWeightThin: 300,
      fontWeightNormal: 300,
      fontWeightMedium: 500,
      fontWeightSemibold: 600,
      fontWeightBold: 700,
    },
    classes: {
      body: {
        fontSize: 'calc(16 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      caption: {
        fontSize: 'calc(12 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      footnote: {
        fontSize: 'calc(14 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      small: {
        fontSize: 'calc(10 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      headline: {
        fontSize: 'calc(24 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      subhead: {
        fontSize: 'calc(18 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      title: {
        fontSize: 'calc(20 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      jumbo: {
        fontSize: 'calc(44 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      mega: {
        fontSize: 'calc(56 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
    },
  },
  components: {
    conversationBox: {
      externalBackgroundColor: 'rgba(94, 124, 139, 0.1)',
      internalBackgroundColor: 'rgba(255, 213, 0, 0.2)',
      importantBackgroundColor: 'rgb(255, 246, 244)',
    },
    loader: {
      backgroundColor: 'rgb(252, 88, 48)',
    },
  },
};

const shallowWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();
  return shallow(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const mountWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const renderWithTheme = (component) => renderer.create(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

const staticWithTheme = (component) => render(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

export {
  defaultTheme,
  shallowWithTheme,
  mountWithTheme,
  staticWithTheme,
  renderWithTheme,
};
