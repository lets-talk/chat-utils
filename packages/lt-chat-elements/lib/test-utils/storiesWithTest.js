import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests({ results: jestTestResults }, {
  filesExt: '__tests__/**.js',
});
