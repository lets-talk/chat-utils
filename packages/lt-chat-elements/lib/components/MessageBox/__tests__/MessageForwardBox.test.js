import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import MessageForwardBox from '../MessageForwardBox';
// Constants
import { renderWithTheme } from '../../../test-utils';

describe('MessageForwardBox component', () => {
  const mockOnForwardClick = jest.fn();
  const message = {
    position: 'left',
  };
  const props = {
    message,
    onForwardClick: mockOnForwardClick,
  };

  it('should render a Text Message and contain the Text and status', () => {
    const component = shallow(<MessageForwardBox {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
    expect(component.find('StyledMessageForwardBox').length).toBe(1);
  });

  test('Styled: Should Render expected Styles when position is left', () => {
    const tree = renderWithTheme(<MessageForwardBox {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
    // background-color: ${(localProps) => rgba(themeColor(localProps.theme, 'foreground', 'base'), 0.1)};
    expect(tree).toHaveStyleRule('right', '-50px');
  });

  test('Styled: Should Render expected Styles when position is right', () => {
    const newProps = {
      message: {
        position: 'right',
      },
      onForwardClick: mockOnForwardClick,
    };
    const tree = renderWithTheme(<MessageForwardBox {...newProps} />).toJSON();
    expect(tree).toMatchSnapshot();
    // background-color: ${(localProps) => rgba(themeColor(localProps.theme, 'foreground', 'base'), 0.1)};
    expect(tree).toHaveStyleRule('left', '-50px');
  });
});
