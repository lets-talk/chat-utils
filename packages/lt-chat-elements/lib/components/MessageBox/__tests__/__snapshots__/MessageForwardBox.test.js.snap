// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForwardBox component Styled: Should Render expected Styles when position is left 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 30px;
  height: 30px;
  border-radius: 20px;
  background: #fff;
  position: absolute;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0 0 5px 0 rgba(164,164,164,1);
  cursor: pointer;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  top: 0;
  bottom: 0;
  margin: auto;
  right: -50px;
}

<div
  className="c0"
  onClick={[MockFunction]}
  onKeyPress={[MockFunction]}
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": "#000",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m40 14.3q0 0.6-0.4 1l-11.5 11.4q-0.4 0.4-1 0.4t-1-0.4-0.4-1v-5.7h-5q-2.2 0-3.9 0.1t-3.4 0.5-3 1-2.4 1.5-1.7 2.3-1.1 3.1-0.4 4q0 1.2 0.1 2.7 0 0.2 0 0.6t0.1 0.6q0 0.3-0.2 0.5t-0.5 0.2q-0.4 0-0.6-0.3-0.2-0.2-0.3-0.5t-0.3-0.7-0.3-0.5q-2.8-6.4-2.8-10.1 0-4.4 1.2-7.4 3.6-9 19.5-9h5v-5.7q0-0.6 0.4-1t1-0.5 1 0.5l11.5 11.4q0.4 0.4 0.4 1z"
      />
    </g>
  </svg>
</div>
`;

exports[`MessageForwardBox component Styled: Should Render expected Styles when position is right 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 30px;
  height: 30px;
  border-radius: 20px;
  background: #fff;
  position: absolute;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0 0 5px 0 rgba(164,164,164,1);
  cursor: pointer;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  top: 0;
  bottom: 0;
  margin: auto;
  left: -50px;
}

<div
  className="c0"
  onClick={[MockFunction]}
  onKeyPress={[MockFunction]}
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": "#000",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m40 14.3q0 0.6-0.4 1l-11.5 11.4q-0.4 0.4-1 0.4t-1-0.4-0.4-1v-5.7h-5q-2.2 0-3.9 0.1t-3.4 0.5-3 1-2.4 1.5-1.7 2.3-1.1 3.1-0.4 4q0 1.2 0.1 2.7 0 0.2 0 0.6t0.1 0.6q0 0.3-0.2 0.5t-0.5 0.2q-0.4 0-0.6-0.3-0.2-0.2-0.3-0.5t-0.3-0.7-0.3-0.5q-2.8-6.4-2.8-10.1 0-4.4 1.2-7.4 3.6-9 19.5-9h5v-5.7q0-0.6 0.4-1t1-0.5 1 0.5l11.5 11.4q0.4 0.4 0.4 1z"
      />
    </g>
  </svg>
</div>
`;

exports[`MessageForwardBox component should render a Text Message and contain the Text and status 1`] = `
<StyledMessageForwardBox
  onClick={[MockFunction]}
  onKeyPress={[MockFunction]}
  position="left"
>
  <FaMailForward
    color="#000"
  />
</StyledMessageForwardBox>
`;
